packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -347,10 +348,11 @@ function safelyDetachRef(current: Fiber, nearestMountedAncestor: Fiber | null) {
 function safelyCallDestroy(
   current: Fiber,
   nearestMountedAncestor: Fiber | null,
-  destroy: () => void,
+  destroy: any => void,
+  resource: {} | void | null,
 ) {
   try {
-    destroy();
+    destroy(resource != null ? resource : undefined);
```

Comment:
Does `destroy(undefined)` do anything? Could lift this out into an if expression if not